### PR TITLE
Feature: Add Auto-Assignment Flow with Result Screen 

### LINKED ISSUE #288: Implement Automatic Task Assignment to Team Members
- [ ] Enable auto-assignment option in TasksScreen
- [ ] Implement algorithm to automatically assign tasks to team members
- [ ] Add confirmation screen when a task is assigned so the user must explicitly accept the assignment before it becomes final

### PR DESCRIPTION
**Pull request by IsmailMoudden**
**Feature: Add Auto-Assignment Flow with Result Screen and Tests**

**Context**

Until now, task assignment in EurekApp required manual input, limiting scalability and fairness in workload distribution.  
This made it difficult for teams to quickly allocate tasks based on availability or project dependencies, resulting in potential inefficiencies and delays in collaborative environments.

This PR introduces a complete **auto-assignment flow**, enabling tasks to be automatically distributed among project members using a hybrid algorithm that balances workload and dependencies.  
The update also includes a dedicated result screen allowing users to review, accept, or apply proposed assignments.

---

**What’s New**

- Implemented `TaskAutoAssignmentService` using a hybrid algorithm (dependency-based + workload-based).  
- Added `AutoAssignResultViewModel` to manage assignment results and user actions (accept/reject/apply).  
- Introduced `AutoAssignResultScreen` composable with interactive UI and Android tests.  
- Integrated auto-assignment entry point within `TasksScreen` and updated navigation routes.  
- Enhanced mock repositories (`MockProjectRepository`, `MockTaskRepository`, `MockUserRepository`) for test coverage.  
- Added unit tests and Android instrumentation tests covering logic, UI behavior, and user interactions.  

---

**User Impact**

- Users can now trigger automatic task assignment with a single action.  
- Tasks are distributed intelligently across members, improving balance and fairness.  
- Clear, interactive interface to review and approve assignment suggestions.  
- Real-time feedback and validation to prevent inconsistencies or failed assignments.  

---

**Implementation Choices**

- **Hybrid assignment algorithm** combining dependency awareness and workload balancing.  
- **Reactive architecture** leveraging `StateFlow` for live updates.  
- **Composable-based UI** for modularity and reuse across task-related screens.  
- **Safe instrumentation tests** built to handle non-deterministic timing in Compose.  

---

**Future Considerations**

- Support multi-project or cross-team auto-assignment.  
- Visualize results with charts or dependency graphs.  
- Integrate machine learning for predictive task allocation.  
- Allow manual fine-tuning of proposed assignments before applying.  

---

**Technical Improvements**

- Shared logic centralized in `ReadWriteTaskViewModel`.  
- Extended Firestore serialization for assignment results.  
- Improved test stability with robust mock repositories.  
- Added safe navigation and error handling patterns across the UI.  

---

**Potential Risks**

- **Assignment fairness** – edge cases in hybrid logic could unbalance workloads.  
  - *Mitigation:* extensive unit tests with randomized workload distributions.  

- **UI synchronization** – asynchronous updates might cause temporary inconsistencies.  
  - *Mitigation:* strict `StateFlow` collection and idle waiting in tests.  

- **Performance impact** for large projects.  
  - *Mitigation:* lightweight algorithm design and early optimization hooks.  

---

**Impact on the Product**

- Introduces a major productivity feature that automates repetitive manual assignments.  
- Enhances collaboration and transparency in project management.  
- Establishes a foundation for future intelligent task automation features.  

*AI was used to assist with parts of the implementation and to help draft the PR description.*

closes #288

### CODE REVIEW

File: app/src/main/java/ch/eureka/eurekapp/ui/tasks/AutoAssignResultViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        if (projects.isEmpty()) {
+          _uiState.value =
+              _uiState.value.copy(
+                  isLoading = false, error = "No projects available for auto-assignment")
+          return@launch
+        }
+
+        val (allTasks, projectTasksMap) = collectTasksFromProjects(projects)
+        val uniqueMembers = collectUniqueMembers(projects)
+
```
Comment (ArluAe) [inline app/src/main/java/ch/eureka/eurekapp/ui/tasks/AutoAssignResultViewModel.kt]: There are multiple repository calls: you first load the projects, then for each project you load the members and tasks, and then you load the user infos. All of this is business logic and should not happen in the ViewModel

File: app/src/main/java/ch/eureka/eurekapp/model/data/StringSerializableEnum.kt [hunk 2]
```diff
@@ -1,21 +1,7 @@
...
 
-/**
- * Interface for enums that can be serialized to/from strings.
- *
- * This interface provides a standard way to convert enum values to display strings (lowercase)
- * while maintaining the default uppercase toString() behavior for Firebase serialization.
- *
- * Note: This file was co-authored by Claude Code.
- */
 interface StringSerializableEnum {
```
Comment (ArluAe) [inline app/src/main/java/ch/eureka/eurekapp/model/data/StringSerializableEnum.kt]: Why did you remove all of the documentation?

File: app/src/main/java/ch/eureka/eurekapp/model/data/chat/ChatChannel.kt [hunk 3]
```diff
@@ -1,18 +1,5 @@
...
- * a subcollection under each channel.
- *
- * Note: This file was co-authored by Claude Code.
- *
- * @property channelID Unique identifier for the chat channel.
- * @property projectId ID of the project this channel belongs to.
- * @property name The name of the chat channel.
- * @property createdBy User ID of the person who created this channel.
- */
 data class ChatChannel(
```
Comment (ArluAe) [inline app/src/main/java/ch/eureka/eurekapp/model/data/chat/ChatChannel.kt]: Why did you remove all of the documentation?

File: app/src/androidTest/java/ch/eureka/eurekapp/model/authentication/AuthRepositoryProviderTest.kt [hunk 4]
```diff
@@ -1,25 +1 @@
-package ch.eureka.eurekapp.model.authentication
```
Comment (ArluAe) [inline app/src/androidTest/java/ch/eureka/eurekapp/model/authentication/AuthRepositoryProviderTest.kt]: Why did you remove this whole test file? It is useful. Please add it back

File: app/src/main/java/ch/eureka/eurekapp/model/data/StringSerializableEnum.kt (hunk 2 above)
Comment (ArluAe) [reply app/src/main/java/ch/eureka/eurekapp/model/data/StringSerializableEnum.kt]: Please add it back

File: app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/AutoAssignResultScreen.kt [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+                        } else {
+                          Text("Apply Selected Assignments ($acceptedCount)")
+                        }
+                      }
+
+                  // Show result message if applied
+                  if (uiState.appliedCount > 0) {
+                    LaunchedEffect(uiState.appliedCount) {
+                      // Navigate back after a short delay
+                      delay(1500)
```
Comment (ArluAe) [inline app/src/main/java/ch/eureka/eurekapp/screens/subscreens/tasks/AutoAssignResultScreen.kt]: Why is there a 1.5s delay? I think you should remove it

**Review by ArluAe — CHANGES_REQUESTED** [review]
Good and useful PR. There are little things to change, but there also changes that you did that I don't understand (like removing doc, adding a delay in the navigation, etc)

Also, make sure that you fix all the sonar issues

**Review by ArluAe — APPROVED** [review]
LGTM. Thanks for applying my fixes. Good job!
